node/src/test/scala/coop/rchain/node/hardening/Hardening.scala

```diff
@@ -0,0 +1,524 @@
+          runtimeManager,
+          preStateHash,
+          validatorPk,
+          seqNum,
+          blockNum
+        ).map {
+          case (postStateHash, chargedDeploysWithMeta, processedDeploys, processedSystem) =>
+            (
+              packBlock(validatorPk, postStateHash, processedDeploys, processedSystem),
+              chargedDeploysWithMeta
+            )
+        }
+      }
+      .toList
+  }
+
+  // simulates 0.99 sync threshold
+  def mkLayer[F[_]: Concurrent: Time: Log](
+      genesisCtx: GenesisContext,
+      baseBlock: BlockMessage,
+      dagStore: BlockDagStorage[F]
+  )(
+      implicit runtimeManager: RuntimeManager[F]
+  ): F[
+    (
+        BlockMessage,
+        Seq[Charged[PaymentDeploy]],
+        List[Seq[Charged[PaymentDeploy]]],
+        List[RejectedDeploy]
+    )
+  ] = {
+
+    val baseState        = baseBlock.body.state.postStateHash
+    val seqNum           = baseBlock.seqNum + 1
+    val mergingBlocksNum = (baseBlock.seqNum * 2 + 1).toLong
+    val mergerBlockNum   = (baseBlock.seqNum * 2 + 2).toLong
+
+    val mkBlocksToMerge = {
+      println(s"Creating ${validatorsNum - 1} concurrent blocks.")
+      Stream
+        .emits(mkBlocks[F](genesisCtx, baseState, seqNum, mergingBlocksNum).tail)
+        .parJoinProcBounded
+        .map {
+          case (b, payment) =>
+            (
+              b.copy(header = b.header.copy(parentsHashList = List(baseBlock.blockHash))),
+              payment
+            )
+        }
+        .evalTap { case (b, _) => dagStore.insert(b, false) }
+        .compile
+        .toList
+    }
+
+    for {
+      // create children blocks
+      t                         <- mkBlocksToMerge
+      (toMerge, mergedPayments) = t.unzip
+
+      // merge children blocks
+      indices <- (baseBlock +: toMerge)
+                  .traverse(
+                    b =>
+                      BlockIndex(
+                        b.blockHash,
+                        b.body.deploys,
+                        b.body.systemDeploys,
+                        Blake2b256Hash.fromByteString(b.body.state.preStateHash),
+                        Blake2b256Hash.fromByteString(b.body.state.postStateHash),
+                        runtimeManager.getHistoryRepo
+                      ).map(b.blockHash -> _)
+                  )
+                  .map(_.toMap)
+      dag <- dagStore.getRepresentation
+      v <- DagMerger.merge[F](
+            dag,
+            baseBlock.blockHash,
+            Blake2b256Hash.fromByteString(baseState),
+            indices(_).deployChains.pure,
+            runtimeManager.getHistoryRepo,
+            DagMerger.costOptimalRejectionAlg
+          )
+      (postState, rejectedDeploys) = v
+      mergedState                  = ByteString.copyFrom(postState.bytes.toArray)
+
+      // create next base block (merge block)
+      _ = println("Creating merge blocks.")
+      r <- mkBlocks[F](genesisCtx, mergedState, seqNum, mergerBlockNum, rejectedDeploys).head
```

Comment:
This creates block on top of merged state - it will serve as a base for the next layer.